runner/config.go

```diff
@@ -383,11 +383,20 @@ func (c *Config) rel(path string) string {
 }
 
 // WithArgs returns a new config with the given arguments added to the configuration.
-func (c *Config) WithArgs(args map[string]TomlInfo) {
-	for _, value := range args {
+func (c *Config) WithArgs(args map[string]TomlInfo) error {
+	for key, value := range args {
 		if value.Value != nil && *value.Value != unsetDefault {
+			// To avoid path resolving problems on Windows.
+			if key == "build.bin" {
```

Comment:
i don't think put logic in there is a good choice.

Reply:
and we need to apply logic to windows only

Reply:
> i don't think put logic in there is a good choice.

I agree on this one, I am thinking of moving the logic inside `setValue2Struct`, wdyt!